api/handler/repo.go

```diff
@@ -1732,21 +1732,19 @@ func (h *RepoHandler) DeployUpdate(ctx *gin.Context) {
 		return
 	}
 
-	var req types.ModelRunReq
+	var req *types.DeployUpdateReq
 	if err := ctx.ShouldBindJSON(&req); err != nil {
 		slog.Error("Bad request format", "error", err, slog.Any("request.body", ctx.Request.Body))
 		httpbase.BadRequest(ctx, err.Error())
 		return
 	}
 
-	if req.Revision == "" {
-		req.Revision = "main" // default repo branch
-	}
-
-	if req.MinReplica < 0 || req.MaxReplica < 0 || req.MinReplica > req.MaxReplica {
-		slog.Error("Bad request setting for replica", slog.Any("MinReplica", req.MinReplica), slog.Any("MaxReplica", req.MaxReplica))
-		httpbase.BadRequest(ctx, "Bad request setting for replica")
-		return
+	if req.MinReplica != nil && req.MaxReplica != nil {
```

Comment:
I don't think we should check if the req.MinReplica is nil, golang should set as 0 by default